src/hypothesis/internal/reflection.py

```diff
@@ -360,14 +360,15 @@ def source_exec_as_module(source):
 
 def accept(%(funcname)s):
     def %(name)s(%(argspec)s):
+        '''%(docstring)s'''
         return %(funcname)s(%(invocation)s)
     return %(name)s
 """.strip() + '\n'
 
 
-def copy_argspec(name, argspec):
-    """A decorator which sets the name and argspec of the function passed into
-    it."""
+def copy_argspec(name, docstring, argspec):
```

Comment:
I'm going to nitpick about naming, sorry: The patch looks fine, but I feel like this is significant scope creep for a function named copy\_argspec.

Given that it's an internal function that is used only in a handful of places I think the best thing to do is probably to rename it. Maybe define\_function\_signature or something suitably generic?